ledger-api/rs-grpc-bridge/src/test/suite/scala/com/digitalasset/grpc/adapter/client/rs/BufferingSubscriptionTest.scala

```diff
@@ -0,0 +1,56 @@
+// Copyright (c) 2022 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.grpc.adapter.client.rs
+
+import com.daml.grpc.adapter.SingleThreadExecutionSequencerPool
+import com.daml.grpc.adapter.client.rs.BufferingSubscriptionTest.MockClientCallStreamObserver
+import io.grpc.stub.ClientCallStreamObserver
+import org.reactivestreams.{Subscriber, Subscription}
+import org.scalatest.concurrent.Eventually.eventually
+import org.scalatest.matchers.should.Matchers
+import org.scalatest.wordspec.AnyWordSpecLike
+
+class BufferingSubscriptionTest extends AnyWordSpecLike with Matchers {
+
+  private val sequencePool = new SingleThreadExecutionSequencerPool("test")
+
+  "Buffering subscription" should {
+
+    "propagate cancellation when no elements were sent" in {
```

Comment:
Since the cancellation mechanism has been significantly changed, could you also add a cancellation test in case an element was sent?

Reply:
Done